### PR TITLE
Add `PasswordField` using `PasswordInput` based on `TextField`

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

cleanshot 2018-12-18 at 13 08 51 2x

A `PasswordInput` based on the updated `TextInput`. 

#### Open Questions

1. Should the field support `isPasswordVisible` from the `PasswordInput`

##### Technical

- Documentation

  - [x] Has `README.md` file
  - [x] Explains use-cases
  - [x] Explains props
  - [x] Has story for Storybook

- Props

  - [x] Uses prop names consistent with existing components
  - [x] Does not use `true` for default-props
  - [x] Does not accept `classNames` prop
  - [x] Has default props for labels (where applicable)

- Design

  - [x] Does not use hard-coded values
  - [x] Uses design-tokens (instead of e.g. color variables)

- Testing

  - [x] Supports `data` attributes
  - [x] Uses [react-testing-library](https://github.com/kentcdodds/react-testing-library) tests
  - [x] Has visual regression tests

- Accessibility (optional)
  - [ ] Supports `aria` attributes

### CODE REVIEW

File: src/components/fields/password-field/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# TextField
```
Comment (emmenko) [inline src/components/fields/password-field/README.md]: PasswordField

File: src/components/fields/password-field/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import { PasswordField } from '@commercetools-frontend/ui-kit';
+
+<PasswordField
+  title="myPassword"
+  value="s3cr3t"
+  onChange={event => alert(event.target.value)}
+/>;
+```
+
+#### Properties
```
Comment (emmenko) [inline src/components/fields/password-field/README.md]: The password input also accepts a `autoComplete` prop

File: src/components/fields/password-field/password-field.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    id: PropTypes.string,
+    horizontalConstraint: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl', 'scale']),
+    errors: PropTypes.shape({
+      missing: PropTypes.bool,
+    }),
+    renderError: PropTypes.func,
+    isRequired: PropTypes.bool,
+    touched: PropTypes.bool,
+
+    // TextInput
```
Comment (emmenko) [inline src/components/fields/password-field/password-field.js]: Password Input

File: src/components/fields/password-field/password-field.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    name: PropTypes.string,
+    value: PropTypes.string.isRequired,
+    onChange: requiredIf(PropTypes.func, props => !props.isReadOnly),
+    onBlur: PropTypes.func,
+    onFocus: PropTypes.func,
+    isAutofocussed: PropTypes.bool,
+    isDisabled: PropTypes.bool,
+    isReadOnly: PropTypes.bool,
+    placeholder: PropTypes.string,
+
```
Comment (emmenko) [inline src/components/fields/password-field/password-field.js]: PS: as mentioned before, the password input also accepts a `autoComplete` prop

**Comment by tdeekens** [comment]
Thanks @emmenko and @dferber90 the last two commits contain the change. Please have another look.

File: src/components/fields/password-field/password-field.form.story.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                'horizontalConstraint',
+                ['xs', 's', 'm', 'l', 'xl', 'scale'],
+                'm'
+              )}
+              errors={formik.errors.password}
+              renderError={key => {
+                switch (key) {
+                  // these could also use <FormattedMessage />
+                  case 'insecure':
+                    return 'Not secure. Use at least one `#`.';
```
Comment (dferber90) [inline src/components/fields/password-field/password-field.form.story.js]: Nice example 👍

**Comment by tdeekens** [comment]
Thanks all!

**Pull request merged**